### PR TITLE
Refactor Repository to use new RepositoryCache class

### PR DESCRIPTION
**Pull request by rspeicher**
Abstracts away the lower-level implementation details from the Repository model.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://tsigo-rs-repository-cache.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://tsigo-rs-repository-cache.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: spec/lib/repository_cache_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'rspec'
+require_relative '../../lib/repository_cache'
```
Comment (tsigo) [inline spec/lib/repository_cache_spec.rb]: Why not just `require 'spec_helper'`? Because it takes *forever* to load if you're just BDD-ing a single class as I was here.

File: lib/repository_cache.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Interface to the Redis-backed cache store used by the Repository model
+class RepositoryCache
+  attr_reader :namespace
+
+  def initialize(namespace, backend = Rails.cache)
```
Comment (tsigo) [inline lib/repository_cache.rb]: Why the second argument? [Dependency Injection](http://en.wikipedia.org/wiki/Dependency_injection)

**Comment by randx** [comment]
@tsigo this looks good. Can you make it mergeable and ping me?

**Comment by tsigo** [comment]
@randx Should be good to go.

**Comment by randx** [comment]
@tsigo thank you

File: lib/repository_cache.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    backend.delete(cache_key(key))
+  end
+
+  def fetch(key, &block)
+    backend.fetch(cache_key(key), &block)
+  end
+
+  private
+
+  attr_reader :backend
```
Comment (randx) [inline lib/repository_cache.rb]: @tsigo why this attr is in end of file?

Comment (tsigo) [inline lib/repository_cache.rb]: I wanted the `attr_reader` for backend but it should be private, which traditionally goes at the bottom. I think you can do `private attr_reader :backend` all on one line and at the top, but that's a bit unusual.

Comment (tsigo) [inline lib/repository_cache.rb]: Actually, after double-checking, that wasn't doing what I thought it was doing. If you want to make it public and move it back to the top that's fine. Either's good.

Comment (randx) [inline lib/repository_cache.rb]: I think we should move it on top. I usually dont scroll to end of file to check available attributes

Comment (tsigo) [inline lib/repository_cache.rb]: Sure thing. Do you want to do that yourself? I think I'd have to submit another PR.

Comment (randx) [inline lib/repository_cache.rb]: @tsigo I dont have too much time for this now so please submit
